### ISSUE TITLE
Imprecise results

### ISSUE THREAD

**Issue by miguelafr** [lauramcastro/readspec#9]
With a QuickCheck state machine that has an operation that defines the following next_state function:

```
create_room_next(S, _R, [RoomId, _Description])
    when RoomId =:= "" -> S;
create_room_next(S, _R, [RoomId, Description])->
    case search_room(RoomId, S#state.rooms) of
        {value, _Room} ->
            S;
        false ->
            S#state {
              rooms = S#state.rooms
            }
    end.
```
apart from the issue reported in #7, there are other things that can not be correct. Thus, this is a piece of code reported by readspec:

```
[*] We define the variable "RoomId" as equal to:
     the 2nd argument passed to create_room/2

```

"RoomId" is the first element of the list, not the second argument of the function create_room.

On the other hand, readspec also reports this:

```
[*] We define the variable "call_to-?MODULE:search/3-1__" as the result provided by
the function called "search" from this same module, when it is
executed with the following 3 arguments:
    - the variable "RoomId"
    - the variable "Rooms"
    - the erlang term: fun (%CTXT-2%Id, %CTXT-2%Room) ->
        %CTXT-2%Room#roomType.roomId == %CTXT-2%Id
end
```

This is not a bug, but it seems readspec uses the implementation of the funcion search_room to build the result:

```
search_room(RoomId, Rooms) ->
    search(RoomId, Rooms,
                fun(Id, Room) -> Room#roomType.roomId == Id end).
```

Is this the expected result? It would be more understandable to use the function search_room in this way:

```
[*] We define the variable "call_to-?MODULE:search_room/2-1__" as the result provided by
the function called "search_room" from this same module, when it is
executed with the following 2 arguments:
    - the variable "RoomId"
    - the variable "Rooms"
end
```

**Comment by palas** [comment]
This is arguable. I am not sure what is better:

I have recently changed it to say something like "the erlang term:" when it
doesn't understand it.

The other option is to just give up the whole function and then it would do
what you say, in this case, but it will also complain if the function that
has the anonymous function is the one you want to analyse.

As it is now, if search_room/2 was in a different module it should also do
what you say, so perhaps it is more versatile.

In summary, this is a less conservative approach, it may sometimes not
explain some terms, but functions can always be abstracted out by moving
them to a different module, so I thought it may be better in practice.

**Comment by miguelafr** [comment]
Ok, moving the function to an external module produces the expected results. Closing the ticket then.
